server/controllers/refer/courseParticipationsController.ts

```diff
@@ -261,10 +263,12 @@ export default class CourseParticipationsController {
           const course = await this.courseService.getCourse(token, courseParticipation.courseId)
           name = course.name
         } else {
-          name = courseParticipation.otherCourseName as CourseParticipationWithName['name']
+          name = courseParticipation.otherCourseName as CourseParticipationPresenter['name']
         }
 
-        return { ...courseParticipation, name }
+        const addedByUser = await this.userService.getUserFromUsername(token, courseParticipation.addedBy)
```

Comment:
I also wonder if it might be worth doing things like this in the (course) service and getting the user details in there and returning the new `addedByName` property as part of `CourseParticipation`. Sorry if this is something that's been discussed already!